### PR TITLE
More test for course use case

### PR DESCRIPTION
**Pull request by HermesCui**
1.Setup basic information in database, one user, one course, and one administer.

2.For the method `registerACourse`, I test 3 cases such as 
- successful creating course
- creating duplicated course
- creating empty course.

3.For the method`removeACourse`, I test 2 cases such as 
- Administer can remove a Course
- Normal User can not remove a Course

4. For the method `modifyCourseContent`, I test 4 cases such as
- Admin can modify content of course
- Normal User can not modify content
- The course can not be found in the database
- The semester can not be modify

5.For the method `addInstructor`, I test 2 cases such as
- Successful adding instructor
- Duplicated instructor is added

6.For the method`removeInstructor`, I test 3 cases such as
- Successful removing instructor
- Normal Use can not remove instructor
- The instructor can not be found

### CODE REVIEW

**Review by Teinble — APPROVED** [review]
Nice! You could state your test results in the content of your pull request next time.

File: src/main/java/use_cases/CourseUseCaseInteractor.java [hunk 1]
```diff
@@ -97,22 +98,11 @@ public void modifyCourseContent(String courseCode, String part, String newPart){
         }
 
         Course course = courseDataInterface.getCourse(courseCode);
-        String[] attributes = {"name", "courseCode", "description", "semester"};
```
Comment (HermesCui) [inline src/main/java/use_cases/CourseUseCaseInteractor.java:100]: Since, the exception updates, the conditions should change.

**Review by blairyeung — COMMENTED** [review]
Good update! I think the update on the UI nad report are essential. 
Have you run the test locally? If all of the test are passed, please enclose a screenshot for further update.
Thank you.

File: src/main/java/use_cases/CourseUseCaseInteractor.java (hunk 1 above)
Comment (HermesCui) [reply src/main/java/use_cases/CourseUseCaseInteractor.java:100]: image

**Review by cang-xue — APPROVED** [review]

**Review by cang-xue — DISMISSED** [review]

**Comment by Teinble** [comment]
This one will be covered by the other PR

**Pull request closed without merging**